src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.Editor.cs

```diff
@@ -0,0 +1,255 @@
+                var membersToRemove = GetMembersToRemove(root, typeNode);
+                foreach (var member in membersToRemove)
+                {
+                    documentEditor.RemoveNode(member, SyntaxRemoveOptions.KeepNoTrivia);
+                }
+
+                var modifiedRoot = documentEditor.GetChangedRoot();
+
+                // add an empty document to solution, so that we'll have options from the right context.
+                var solutionWithNewDocument = projectToBeUpdated.Solution.AddDocument(newDocumentId, newDocumentName, text: string.Empty);
+
+                // update the text for the new document
+                solutionWithNewDocument = solutionWithNewDocument.WithDocumentSyntaxRoot(newDocumentId, modifiedRoot, PreservationMode.PreserveIdentity);
+
+                // get the updated document, perform clean up like remove unused usings.
+                var newDocument = solutionWithNewDocument.GetDocument(newDocumentId);
+                return await CleanUpDocumentAsync(newDocument, cancellationToken).ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// update the original document and remove the type that was moved.
+            /// perform other fix ups as necessary.
+            /// </summary>
+            /// <param name="sourceDocument">original document</param>
+            /// <param name="typeNode">type that was moved to new document</param>
+            /// <param name="cancellationToken">a cancellation token</param>
+            /// <returns>an updated solution with the original document fixed up as appropriate.</returns>
+            private async Task<Solution> UpdateSourceDocumentAsync(
+                Document sourceDocument, TTypeDeclarationSyntax typeNode, CancellationToken cancellationToken)
+            {
+                var documentEditor = await DocumentEditor.CreateAsync(sourceDocument, cancellationToken).ConfigureAwait(false);
+
+                AddPartialModifiersToTypeChain(documentEditor, typeNode);
+                documentEditor.RemoveNode(typeNode, SyntaxRemoveOptions.KeepNoTrivia);
+
+                var updatedDocument = documentEditor.GetChangedDocument();
+                return await CleanUpDocumentAsync(updatedDocument, cancellationToken).ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// Traverses the syntax tree of the forked document and
+            /// collects a list of nodes that are not being moved.
+            /// This list of nodes are then removed from the forked copy.
+            /// </summary>
+            /// <param name="root">root, of the syntax tree of forked document</param>
+            /// <param name="typeNode">node being moved to new document</param>
+            /// <returns>list of syntax nodes, to be removed from the forked copy.</returns>
+            private static IEnumerable<SyntaxNode> GetMembersToRemove(
+                SyntaxNode root, TTypeDeclarationSyntax typeNode)
```

Comment:
1. Get the parent chain of typeNode and add that to a HashSet
2. call Root.DescendentNodes()

the descendIntoChildren parameter will be a function taht returns true as long as syntaxNode != typeNode.
FOr every node you hit, you want to add to the list of nodes to remove as long as it isn't in the parent chain set.

Also, don't remove imports.